crates/kernel/src/tasks/task.rs

```diff
@@ -211,39 +211,41 @@ impl Task {
             let control_msg = if task.vm_host.is_running() {
                 match task_control_receiver.try_recv() {
                     Ok(msg) => msg,
-                    Err(_) => {
+                    Err(TryRecvError::Disconnected) => {
                         warn!(task_id = ?task.task_id, "Channel closed");
                         task.done = true;
 
                         break;
                     }
+                    Err(TryRecvError::Empty) => {
+                        // No message, just keep going.
+                        continue;
```

Comment:
This almost looks like a busy wait loop, except: we know the channel has non-zero capacity (because we create unbounded channels), and so `try_recv()` will actually block. Confirm?

Reply:
Erm, yeah, what is happening here (or supposed to be) is:

If the VM is actively executing a method, don't block.  Just check for a message then return immediately back into the loop to fetch and run the next opcodes.
Otherwise, block waiting for a message.

This way we can still respond to messages while executing opcodes in the interpreter loop.

I have this chunk mentally flagged because I feel like it could be cleaned up with condition variables or atomic flags replacing channels. In reality most of the stuff the scheduler is "asking" the task to do could just be AtomicBools or something that are checked inside the VM execution loop.